### ISSUE TITLE
Java architecture is really procedural

### ISSUE THREAD

**Issue by amihaiemil** [yegor256/jpeek#183]
The Jaba part of Jpeek is really procedural. Classes like ``Reports`` and ``App`` are filled with ``if`` statements and maps of parameters called ``params``. 

I saw the effects of this in 3 PRs so far: https://github.com/yegor256/jpeek/pull/168, https://github.com/yegor256/jpeek/pull/173 and https://github.com/yegor256/jpeek/pull/179

All 3 of them introduced rather ugly code, but it's not their fault, that's where the architecture drives to. I think we should start refactoring.

We can start with a ``PomXml`` object which would animate the base's ``pom.xml`` file and provide any info related to it.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by llorllale** [comment]
I think I'm OK with just naming it `Pom`:

```java
public final class Pom implements XML {
   //override XML methods
   //add new methods specific to pom.xml
}
```

**Comment by amihaiemil** [comment]
@llorllale Not really, if it implements ``XML`` then it will offer just the "low level" operations of interface XML. I would like it to be more high level and have methods such as ``version()``, ``scmUrl()``, ``developers()``, ``dependencies()`` etc

**Comment by llorllale** [comment]
@amihaiemil agreed

Although I prefer the name `Pom` instead of `PomXml`.
